File: .agent/skills/rb-ghl-automation/scripts/common_ghl.js
const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, '.env') });
const axios = require('axios');

const BASE_URL = "https://services.leadconnectorhq.com";

function getHeaders() {
    const token = process.env.GHL_PRIVATE_INTEGRATION_TOKEN;
    const locationId = process.env.GHL_LOCATION_ID;

    if (!token) {
        throw new Error("Missing GHL_PRIVATE_INTEGRATION_TOKEN in .env file.");
    }

    const headers = {
        "Authorization": `Bearer ${token}`,
        "Version": "2021-07-28",
        "Content-Type": "application/json",
        "Accept": "application/json"
    };

    if (locationId) {
        headers["locationId"] = locationId;
    }

    return headers;
}

async function makeRequest(method, endpoint, params = {}, data = {}) {
    const url = `${BASE_URL}${endpoint}`;
    const headers = getHeaders();

    // Auto-inject locationId into params for GET requests if missing
    if (method === 'GET' && !params.locationId && headers.locationId) {
        params.locationId = headers.locationId;
    }

    try {
        const config = {
            method: method,
            url: url,
            headers: headers,
            params: params,
            data: data
        };

        const response = await axios(config);
        return response.data;
    } catch (error) {
        if (error.response) {
            console.error(`HTTP Error: ${error.response.status} - ${error.response.statusText}`);
            console.error(`Response Data:`, JSON.stringify(error.response.data, null, 2));
        } else {
            console.error(`Error: ${error.message}`);
        }
        return null;
    }
}

module.exports = { makeRequest };

File: .agent/skills/rb-ghl-automation/scripts/common_ghl.py
import os
import requests
import json
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

BASE_URL = "https://services.leadconnectorhq.com"

def get_headers():
    token = os.getenv("GHL_PRIVATE_INTEGRATION_TOKEN")
    location_id = os.getenv("GHL_LOCATION_ID")
    
    if not token:
        raise ValueError("Missing GHL_PRIVATE_INTEGRATION_TOKEN in environment variables.")
    
    headers = {
        "Authorization": f"Bearer {token}",
        "Version": "2021-07-28",
        "Content-Type": "application/json",
        "Accept": "application/json"
    }
    
    # Some endpoints require locationId in headers, others in query params.
    # We add it to headers just in case, but specific scripts should add it to params if needed.
    if location_id:
        headers["locationId"] = location_id
        
    return headers

def make_request(method, endpoint, params=None, data=None):
    url = f"{BASE_URL}{endpoint}"
    headers = get_headers()
    
    try:
        response = requests.request(method, url, headers=headers, params=params, json=data)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.HTTPError as err:
        print(f"HTTP Error: {err}")
        print(f"Response Body: {response.text}")
        return None
    except Exception as err:
        print(f"Error: {err}")
        return None
